corehq/blobs/tests/test_migrate.py

```diff
@@ -0,0 +1,60 @@
+# coding=utf-8
+import json
+from os.path import join
+
+import corehq.blobs.migrate as mod
+from corehq.blobs.mixin import BlobMixin
+from couchexport.models import SavedBasicExport, ExportConfiguration
+from django.test import TestCase
+from testil import assert_raises, eq, tempdir
+
+
+class TestMigrations(TestCase):
+
+    def test_migrate_saved_exports(self):
+        # TODO generalize this to make it reusable for other slugs
+        slug = "saved_exports"
+
+        def _mk_config(name='some export name', index='dummy_index'):
+            return ExportConfiguration(index=index, name=name, format='xlsx')
+
+        # setup data
+        saved = SavedBasicExport(configuration=_mk_config())
+        saved.save()
+        payload = 'something small and simple'
+        name = saved.get_attachment_name()
+        super(BlobMixin, saved).put_attachment(payload, name)
+        saved.save()
+
+        # verify: attachment is in couch and migration not complete
+        eq(len(saved._attachments), 1)
+        eq(len(saved.external_blobs), 0)
+        with assert_raises(mod.BlobMigrationState.DoesNotExist):
+            mod.BlobMigrationState.objects.get(slug=slug)
+
+        with tempdir() as tmp:
+            filename = join(tmp, "file.txt")
+
+            # do migration
+            migrated = mod.MIGRATIONS[slug].migrate(filename)
+            assert migrated >= 1, migrated
+
+            # verify: migration state recorded
+            mod.BlobMigrationState.objects.get(slug=slug)
+
+            # verify: migrated data was written to the file
+            with open(filename) as fh:
+                lines = list(fh)
+            doc = {d["_id"]: d for d in (json.loads(x) for x in lines)}[saved._id]
+            eq(doc["_rev"], saved._rev)
+            eq(len(lines), migrated, lines)
+            data = doc["_attachments"].values()[0]["data"].decode("base64")
+            eq(data, payload)
+
+        # reload and verify: attachment was moved to blob db
+        exp = SavedBasicExport.get(saved._id)
+        assert exp._rev != saved._rev, exp._rev
+        eq(len(exp.blobs), 1)
+        assert not exp._attachments, exp._attachments
+        eq(len(exp.external_blobs), 1)
+        eq(exp.get_payload(), payload)
```

Comment:
I'm not suggesting we change all tests to use this style, but I think they make writing and debugging tests easier.

- `eq` - nicer output when the things being compared are medium-length strings. This is an adaptation of a similar function from [`nose.tools`](http://nose.readthedocs.org/en/latest/testing_tools.html).
- `assert_raises` - works both as a function `assert_raises(Exc, func, args)` and context manager `with assert_raises(Exc): ...`. The context manager has a powerful `msg` parameter that can be used to make assertions about the raised exception beyond its type. See [`testil.assert_raises`](https://github.com/millerdev/testil/blob/master/testil.py#L191-L256)
- `assert ...` is short, clear and works well when a more powerful helper is not needed to introspect the assertion parameters.

Do you think it's not a good idea to use these?